react/__tests__/Slider.test.tsx

```diff
@@ -0,0 +1,353 @@
+    const paginationDots = getByLabelText('Slider pagination dots')
+
+    expect(expectedCarouselRole).toBeDefined()
+    expect(leftArrow).toBeDefined()
+    expect(rightArrow).toBeDefined()
+    expect(paginationDots).toBeDefined()
+  })
+
+  it('should render without pagination features and use x-scrolling if usePagination is set to false', () => {
+    const { getByTestId, queryByTestId, queryByLabelText } = render(
+      <Slider
+        totalItems={10}
+        itemsPerPage={5}
+        showNavigationArrows="always"
+        showPaginationDots="always"
+        usePagination={false}
+        arrowSize={25}
+        fullWidth
+        infinite
+      />
+    )
+
+    const sliderTrackContainer = getByTestId('slider-track-container')
+    const leftArrow = queryByTestId('icon-caret-left')
+    const rightArrow = queryByTestId('icon-caret-right')
+    const paginationDots = queryByLabelText('Slider pagination dots')
+
+    expect(leftArrow).toBeNull()
+    expect(rightArrow).toBeNull()
+    expect(paginationDots).toBeNull()
+    expect(sliderTrackContainer.className.split(' ')).toContain(
+      'overflow-x-scroll'
+    )
+  })
+
+  it('should render without pagination features if there are not enough slides to fill a page', () => {
+    const { getByTestId, queryByTestId, queryByLabelText } = render(
+      <Slider
+        totalItems={3}
+        itemsPerPage={5}
+        showNavigationArrows="always"
+        showPaginationDots="always"
+        arrowSize={25}
+        usePagination
+        fullWidth
+        infinite
+      />
+    )
+
+    const sliderTrackContainer = getByTestId('slider-track-container')
+    const leftArrow = queryByTestId('icon-caret-left')
+    const rightArrow = queryByTestId('icon-caret-right')
+    const paginationDots = queryByLabelText('Slider pagination dots')
+
+    expect(leftArrow).toBeNull()
+    expect(rightArrow).toBeNull()
+    expect(paginationDots).toBeNull()
+
+    // Make sure that the slider renders without x-scrolling, since usePagination is set to true
+    expect(sliderTrackContainer.className.split(' ')).not.toContain(
+      'overflow-x-scroll'
+    )
+  })
+
+  it('should add appropriate padding in the x-axis if slider is not a full width one', () => {
+    const { getByLabelText } = render(
+      <Slider
+        totalItems={10}
+        itemsPerPage={5}
+        showNavigationArrows="always"
+        showPaginationDots="always"
+        fullWidth={false}
+        arrowSize={25}
+        usePagination
+        infinite
+      />
+    )
+
+    const sliderSectionElement = getByLabelText('slider')
+    const { paddingLeft } = sliderSectionElement.style
+    const { paddingRight } = sliderSectionElement.style
+
+    expect(paddingLeft).toEqual('50px')
+    expect(paddingRight).toEqual('50px')
+  })
+})
+
+describe('Behavior upon interaction', () => {
+  it('should correctly toggle arrows and pagination dots visibility on mobile devices', () => {
```

Comment:
Maybe this test should be three? 🤔

Reply:
I think it makes sense for these assertions to be part of the same test. They're testing the same behavior, which is whether or not pagination dots and arrows visibility would be resolved to the correct values when on mobile devices.

This is basically testing this bit of code:

```typescript

const shouldShowArrows = Boolean(
    (showNavigationArrows === 'always' ||
      (showNavigationArrows === 'mobileOnly' && isMobile) ||
      (showNavigationArrows === 'desktopOnly' && !isMobile)) &&
      !shouldBeStaticList
  )

  const shouldShowPaginationDots = Boolean(
    (showPaginationDots === 'always' ||
      (showPaginationDots === 'mobileOnly' && isMobile) ||
      (showPaginationDots === 'desktopOnly' && !isMobile)) &&
      !shouldBeStaticList
  )
```

with `isMobile` being a constant value.